spec/diagnose_spec.rb

```diff
@@ -347,12 +565,38 @@
       ]
     )
   end
+
+  it "does not submit a report" do
+    expect(@received_report).to be_nil
+  end
+end
+
+RSpec.describe "Running the diagnose command with the --send-report option" do
+  before :all do
+    @runner = init_runner(:args => ["--send-report"])
+    @runner.run
+    @received_report = DiagnoseServer.last_received_report
+  end
+
+  it "sends the report automatically" do
+    send_report = section(:send_report)
+    expect(send_report).to_not include("Send diagnostics report to AppSignal?")
+    expect(send_report).to include(
+      "  Your support token: diag_support_token",
+      "  View this report: https://appsignal.com/diagnose/diag_support_token"
+    )
+  end
+
+  it "submit a report automatically" do
+    expect(@received_report).to be_instance_of(DiagnoseReport)
```

Comment:
It would be good to check somewhere (not sure if here's the right place) that the `api_key` param is sent along when a report is submitted, to cover the fix for its absence at appsignal/appsignal-nodejs#474.

Additionally, it could be good to check that only one report is received, more as a "sanity check" than anything.

Reply:
Yes, that would be a good thing to test as well.